odl/discr/discr_utils.py

```diff
@@ -1332,33 +1353,18 @@ def dual_use_func(x, out=None, **kwargs):
             elif tensor_valued:
                 # The out object can be any array-like of objects with shapes
                 # that should all be broadcastable to scalar_out_shape.
-                results = np.array(out)
-                if results.dtype == object or scalar_in:
-                    # Some results don't have correct shape, need to
-                    # broadcast
-                    bcast_res = []
-                    for res in results.ravel():
-                        if ndim == 1:
-                            # As usual, 1d is tedious to deal with. This
-                            # code deals with extra dimensions in result
-                            # components that stem from using x instead of
-                            # x[0] in a function.
-                            # Without this, broadcasting fails.
-                            shp = getattr(res, 'shape', ())
-                            if shp and shp[0] == 1:
-                                res = res.reshape(res.shape[1:])
-                        bcast_res.append(
-                            np.broadcast_to(res, scalar_out_shape))
+                try:
+                    out_arr = np.asarray(out)
+                except ValueError:
```

Comment:
Previously there was a check if scalar_in ==True. Now it is gone. I can't see why it was necessary....

Reply:
Could it be because of the corner case when ndim=1?

Reply:
Honestly I can't see why it was necessary in the old version either, but the new check `np.isscalar(arr_lists)` in line 930 certainly makes sure that scalars are handled correctly.